consensus/src/consensus_observer/network/observer_message.rs

```diff
@@ -336,6 +336,38 @@ impl PayloadWithProof {
     }
 }
 
+/// The transaction payload and proof of each block with a transaction and block gas limit
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PayloadWithProofAndLimits {
+    payload_with_proof: PayloadWithProof,
+    transaction_limit: Option<u64>,
+    gas_limit: Option<u64>,
```

Comment:
should we make this an enum, so we just need to extend this in future if necessary?

Reply:
I was thinking this, but thought we probably don't need to extend it. Probably what we thought before too :P

Reply:
@ibalajiarun changed to an enum. There's a lot of "similar-looking and sounding" structs in here now, that we'll need to clean up later